### PR TITLE
Http response

### LINKED ISSUE #4: Use HTTP Status Codes Consistently
**Description of the current situation**
Currently, the api returns to the frontend a game state where there is a `:status`. The current *status* is only used by the frontend to decide what to show, but can be known using only completely public information.

Also, at `configurations/configuration.yml` and the files that use that we are treating the same way the messages used to describe the statuses and errors.

And (probably the most notorious thing) we are always returning code 200, even when we return an error message.

**Description of what I'd like**
A clear documentation on how we use HTTP statuses and create a wrapper function that assigns it with the appropiate code and message. An important part would be returning code 4xx when we are also returning an error message.

That would imply modifying the `configurations/configuration.yml` so it intuitively shows which are error messages and which aren't

**Describe alternatives you've considered**
I've thought about simply dividing the `configuration.yml` file in statuses and errors, but that doesn't makes thinks clearer, since we are always returning a status code 200.

Also, I created this Issue suggesting to remove the `:status` from backend, but apparently that wouldn't be compliant with HTTP status codes.

**Notes**
I labeled it as minor, but if we begin to use this status for more things (probably for weird or lazy reasons) it'll be more difficult to change.

### LINKED ISSUE #9: Exceptions should never reach the client
**Description**
When encountering exceptions such as produced by #8, the client receives all of the exception data. This should never happen, we should hide any exceptions from the client and instead log them while returning an error code of the 50x class (Internal Server Error)

**To Reproduce**
See #8 

**Expected behavior**
A normal JSON response with `:status 50x`

**Additional context**
This is related to #4, but due to the added layer of catching server exceptions it's a separate issue.

### PR DESCRIPTION
**Pull request by Masclins**
**Description**

This Pull Requests comes from the branch on #19 

It changes the way we return RESTful responses so they are all with the form
```
{
  :headers {}
  :status xxx
  :body {...}
}
```

**Issues Resolved**

Resolves #4 and resolves #9 (I found it easy to include it... though it might require some changes. Could discuss it and divide it in two different PR)

**Details**

All messages with an `:error` on the body will trigger a 400 status. That includes "page not found", which right now will return a 400 status too.
All messages that raise an exception will return a 500 status and no body.

**Hotspots**

It's worth thinking if we'd need any specific status, like 403 (forbidden), 404 (not found), 405 (method not allowed)...

Also, we really should try to implement the pacts, but I've no idea how to do that.

**Checklist for contribution requirements**

- [x] The tests pass
- [ ] The code has tests
- [ ] The tests are well-designed
- [ ] The code is readable
- [ ] The code is well-organized
- [ ] Any rule changed or included is also on the Manual

### CODE REVIEW

File: src/api/handler.clj [hunk 1]
```diff
@@ -22,31 +18,26 @@
...
-                   (api/get-game (parse-int id) player))
+                   {:body (api/get-game (parse-int id) player)})
     (compojure/POST "/games/:id{[0-9]+}/player/:player"
                     [id player :as {body :body}]
-                    (play-action (parse-int id) player body))
-    (route/not-found "<h1>Page not found</h1>")))
+                    {:body (play-action (parse-int id) player body)})
+    (route/not-found {:body {:error "Page not found"}})))
 
-(def ^:private cors-headers
```
Comment (kenan-rhoton) [inline src/api/handler.clj:37]: removing CORS compliance will give us problems for a Browser-Based CLI :cry:

**Review by kenan-rhoton — COMMENTED** [review]
Your changes get mixed up with the persistence-disk changes, so I'll wait for that to get merged to get a clearer picture of exactly what this entails

File: src/api/handler.clj (hunk 1 above)
Comment (Masclins) [reply src/api/handler.clj:37]: I'll add them again.

**Comment by kenan-rhoton** [comment]
Ok, this seems perfect, now we just have to decide on a way to actually test it... (at least the 400 errors, I don't think the 500 is actually testable since it would imply us having a bug in the code and the test would fail whenever we fixed the bug)

**Review by kenan-rhoton — APPROVED** [review]
Looks perfect! Thanks a lot! We might expand the api tests further using this, but I would make a separate issue for that (and that way we can have nothing ignored by cloverage and yagni, which is amazing)

**Pull request merged**